### PR TITLE
feature/add-i18n

### PR DESCRIPTION
**Pull request by francochau**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "i18n-ally.localesPaths": ["public/locales"]
```
Comment (harry830622) [inline .vscode/settings.json:2]: Please don't add vscode related configs into the repo

File: pages/_app.js [hunk 2]
```diff
@@ -97,19 +98,24 @@ const MyApp = ({ Component, pageProps }) => {
...
+    ) {
+      ref.current.emContract = new ethers.Contract(
+        process.env.NEXT_PUBLIC_EM_ADDRESS,
+        emAbi,
+        ref.current.provider,
+      );
+      ref.current.mergeContract = new ethers.Contract(
+        process.env.NEXT_PUBLIC_MERGE_ADDRESS,
+        mergeAbi,
+        ref.current.provider,
```
Comment (harry830622) [inline pages/_app.js:116]: These env variables are not optional, so just leave it as is before.

File: .vscode/settings.json (hunk 1 above)
Comment (francochau) [reply .vscode/settings.json]: @harry830622 removed

File: pages/_app.js (hunk 2 above)
Comment (francochau) [reply pages/_app.js:116]: Okay, but you may need to pass the .env to me and vercel staging environment later on

Comment (harry830622) [reply pages/_app.js:116]: Sure, let me DM you in the discord

File: pages/_app.js [hunk 3]
```diff
@@ -97,7 +98,7 @@ const MyApp = ({ Component, pageProps }) => {
     if (!ref.current.provider) {
       ref.current.provider = new ethers.providers.AlchemyProvider(
         process.env.NEXT_PUBLIC_NETWORK,
-        process.env.NEXT_PUBLIC_RPC_URL.split('/').pop(),
+        process.env.NEXT_PUBLIC_RPC_URL?.split('/').pop(),
```
Comment (harry830622) [inline pages/_app.js:101]: You forgot here 🤣

Comment (francochau) [reply pages/_app.js:101]: damn lol

**Comment by harry830622** [comment]
Perfect! Really appreciate your help.

**Pull request merged**